src/objects.c

```diff
@@ -1333,9 +1623,9 @@ static void unlink_server(PgSocket *server, const char *reason)
  * The latter is for protocol and communication errors where a normal
  * protocol termination is not possible.
  */
-void disconnect_server(PgSocket *server, bool send_term, const char *reason, ...)
+void disconnect_server(PgSocket *server, bool send_term, int thread_id, const char *reason, ...)
```

Comment:
One thing I dislike in the PR is this infection of `thread_id` everywhere througout the API. My understanding is that each `PgSocket` is essentially bound to a thread, because its `SBuf` is bound to a thread. It seems much nicer to have the `thread_id` be extracted from this `SBuf` whenever it's needed when a `PgSocket` is already part of the arguments, instead of then separately passing the `thread_id` again. This will also make the PR much easier to review, because it will have much fewer changed lines of code.

If for some reason the thread_id from the `SBuf` cannot be used for this, it still seems much better to put the `thread_id` of the `PgSocket` in the `PgSocket` definition, so we can still remove these arguments.

Reply:
I think we should definitely try to avoid it at the serven/client level functions too. It seems that all this is only/mostly needed for `disconnect_client` and `disconnect_server`. Another approach would be to have a variant of disconnect_server that takes a thread_id for the places where that's actually needed, but for the normal case it would use the thread id of the server (so have two functions `disconnect_server` and `disconnect_server_on_thread`).